linker-interface/shared/src/test/scala/org/scalajs/linker/interface/StandardConfigTest.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.interface
+
+import org.junit.Test
+import org.junit.Assert._
+import org.junit.function.ThrowingRunnable
+
+class StandardConfigTest {
+  @Test
+  def jsHeader(): Unit = {
+    def testValid(header: String): Unit = {
+      assertTrue(header, StandardConfig.isValidJSHeader(header))
+      assertEquals(header, StandardConfig().withJSHeader(header).jsHeader)
+    }
+
+    def testInvalid(header: String): Unit = {
+      assertFalse(header, StandardConfig.isValidJSHeader(header))
+      assertThrows(classOf[IllegalArgumentException], new ThrowingRunnable {
+        def run(): Unit = StandardConfig().withJSHeader(header)
```

Comment:
Optional: Maybe just enable SAM support for 2.11?

Reply:
So far we've avoided doing that in code that can get away without it. The library is special because it *needs* the SAM handling (without it, the Scala to JS functions conversions would go into an infinite loop).